metadata-ingestion/src/datahub/ingestion/source/cassandra/cassandra_config.py

```diff
@@ -0,0 +1,189 @@
+import datetime
+from typing import List, Optional
+
+import pydantic
+from pydantic import Field
+
+from datahub.configuration.common import AllowDenyPattern, ConfigModel
+from datahub.configuration.source_common import (
+    EnvConfigMixin,
+    PlatformInstanceConfigMixin,
+)
+from datahub.ingestion.source.ge_profiling_config import GEProfilingConfig
+from datahub.ingestion.source.state.stale_entity_removal_handler import (
+    StatefulStaleMetadataRemovalConfig,
+)
+from datahub.ingestion.source.state.stateful_ingestion_base import (
+    StatefulIngestionConfigBase,
+)
+from datahub.ingestion.source_config.operation_config import is_profiling_enabled
+
+# - Referencing https://docs.datastax.com/en/cql-oss/3.x/cql/cql_using/useQuerySystem.html#Table3.ColumnsinSystem_SchemaTables-Cassandra3.0 - #
+# this keyspace contains details about the cassandra cluster's keyspaces, tables, and columns
+SYSTEM_SCHEMA_KEYSPACE_NAME = "system_schema"
+
+# Reference:
+# https://docs.datastax.com/en/astra-db-serverless/databases/python-driver.html
+# https://docs.datastax.com/en/astra-db-serverless/databases/python-driver.html#production-configuration
+
+
+class CassandraCloudConfig(ConfigModel):
+    """
+    Configuration for connecting to DataStax Astra DB in the cloud.
+    """
+
+    token: str = Field(
+        description="The Astra DB application token used for authentication.",
+    )
+
+    secure_connect_bundle: str = Field(
+        description="File path to the Secure Connect Bundle (.zip) used for a secure connection to DataStax Astra DB.",
+    )
+
+    connect_timeout: int = Field(
+        default=600,
+        description="Timeout in seconds for establishing new connections to Cassandra.",
+    )
+
+    request_timeout: int = Field(
+        default=600, description="Timeout in seconds for individual Cassandra requests."
+    )
+
+
+class ProfileConfig(GEProfilingConfig):
```

Comment:
What I really meant is to have ProfileConfig defined in some common place such that it contains only the relevant most common subset of fields along with description. To avoid duplication,  GEProfilingConfig can use that one as base class but thats not neccessary